### PR TITLE
aggregated bar chart

### PR DESCRIPTION
**Pull request by MaximKomlev**
The main idea of the new chart (it is based on bar chart) is to add possibility visually aggregate rendered bars under single bar, e.g. for such data, like temperature changes during some period of time, it is usefull visually agregate the bars under one bar, and show extrime values (min/max or median), suggest that bars have constant size and can not be scaled, so scalling of the timeline will discloses aggregated values. For now, the bars are aggreagted under max value of data entries which were aggreageted. Perhaps, you will find it usefull.

### CODE REVIEW

**Comment by jjatie** [comment]
Please provide images in your description and a demo in your PR (iOS or Mac OS, preferably in Swift).

**Comment by MaximKomlev** [comment]
Hi @jjatie, I attached couple screenshots made on iPad, also the pull request has a sample under ChartsDemo, AggregateBarChartViewController.*
aggregated_bar_chart1
aggregated_bar_chart2

is it what you mentioned in your comment?

**Comment by MaximKomlev** [comment]
hi @jjatie, I added images from real app, first shows default state, second if user scales timeline.
default
zoomed

**Comment by liuxuan30** [comment]
sorry, I don't understand what's aggregated.. all images looks just like normal bar charts..

**Comment by MaximKomlev** [comment]
Idea was to stay with the same width of bar, and on scale of timeline, place the bar in appropriate place on timeline, if there is interaction of bars since width static then aggregate them into one.
I attached screenshot of default bar chart with min scale of timeline and when scaling to middle (another_bar_chart_min_timelinescale and another_bar_chart_middle_timlenescale).

and to see difference attached aggregated bar chart width the same state (aggregated_bar_chart_min_timelinescale and aggregated_bar_chart_middle_timelinescale)

another_bar_chart_min_timelinescale and another_bar_chart_middle_timlenescale:
another_bar_chart_middle_timlenescale
another_bar_chart_min_timelinescale

aggregated_bar_chart_min_timelinescale and aggregated_bar_chart_middle_timelinescale:
aggregated_bar_chart_middle_timelinescale
aggregated_bar_chart_min_timelinescale

**Comment by liuxuan30** [comment]
Are you saying the bar width can keep the same not matter how much you zoomed in?

**Comment by MaximKomlev** [comment]
yes, it keeps width and position on timeline.

**Comment by liuxuan30** [comment]
I see. 

current master won't accept feature PRs  right now, as we are trying to fix some issues and release 3.1

Please switch the branch to 4.0 and fix any conflicts.

Currently 4.0 branch contains our swift refactoring, so please keep the code style consistent.

File: Source/Charts/Renderers/AggregatedBarChartRenderer.swift [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+import Foundation
+import CoreGraphics
+
+#if !os(OSX)
+    import UIKit
+#endif
+
+
+// TODO: handle multiple datasets
```
Comment (petester42) [inline Source/Charts/Renderers/AggregatedBarChartRenderer.swift]: should handle multiple datasets

**Pull request closed without merging**